src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -75,7 +75,168 @@ module LeafExpressionConverter =
 
```

Comment:
The changes in this file are very extensive.  I trtust you that bugs are being fixed, but the title of the PR currently says  "Subtraction of two chars" and the description "Fixes fsharp/fslang-suggestions#1030"

Could you please expand out the title and description of the PR extensively - they need to convey everything that is being fixed here.  Also we have to assess everything w.r.t. `/langversion`